addons/mail/static/src/core/common/thread_model.js

```diff
@@ -1000,7 +1025,7 @@ export class Thread extends Record {
 
     addOrReplaceMessage(message, tmpMsg) {
         // The message from other personas (not self) should not replace the tmpMsg
-        if (tmpMsg && tmpMsg.in(this.messages) && message.author.eq(this.store.self)) {
+        if (tmpMsg && tmpMsg.in(this.messages) && this.matchesSelfIdentity(message.author)) {
```

Comment:
In this case, I believe it is also related to who does the action, and not matching all.
But all of this is necessary because there can be conflicting tmp ids coming from different users.
If temp id properly used uuid none of this would be a problem.